services/src/main/java/org/fao/geonet/api/pages/PagesAPI.java

```diff
@@ -0,0 +1,475 @@
+    public void addPage(
+            @RequestParam(value = "language", required=true) final String language,
+            @RequestParam(value = "pageId", required=true) final String pageId,
+            @RequestParam(value = "data", required=false) MultipartFile data,
+            @RequestParam(value = "format", required=true) final Page.PageFormat format,
+            @ApiIgnore final HttpServletResponse response
+            ) {
+
+        final ApplicationContext appContext = ApplicationContextHolder.get();
+        PageRepository pageRepository = appContext.getBean(PageRepository.class);
+
+        if(pageRepository.findOne(new PageIdentity(language, pageId)) == null) {
+
+            List<Page.PageSection> sections = new ArrayList<Page.PageSection>();
+            sections.add(Page.PageSection.DRAFT);
+            byte[] bytesData;
+            try {
+                bytesData = data.getBytes();
+            } catch (Exception e) {
+                response.setStatus(HttpStatus.CONFLICT.value());
+                return;
+            }
+
+            Page page = new Page(new PageIdentity(language, pageId), bytesData, format, sections, Page.PageStatus.HIDDEN);
+
+            pageRepository.save(page);
+        } else {
+            response.setStatus(HttpStatus.CONFLICT.value());
+        }
+    }
+
+    // Isn't done with PUT because the multipart support as used by Spring
+    // doesn't support other request method then POST
+    @ApiOperation(
+            value = "Edit a Page object",
+            notes = "<a href='http://geonetwork-opensource.org/manuals/trunk/eng/users/user-guide/define-static-pages/define-pages.html'>More info</a>",
+            nickname = "editPage")
+    @RequestMapping(
+            value = "/{language}/{pageId}",
+            method = RequestMethod.POST)
+    @ResponseStatus(value = HttpStatus.OK)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = PAGE_UPDATED),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    @PreAuthorize("hasRole('Administrator')")
+    public void editPage(
+            @PathVariable(value = "language") final String language,
+            @PathVariable(value = "pageId") final String pageId,
+            @RequestParam(value = "data", required=false) MultipartFile data,
+            @RequestParam(value = "format", required=true) final Page.PageFormat format,
+            @ApiIgnore final HttpServletResponse response
+            ) {
+        final ApplicationContext appContext = ApplicationContextHolder.get();
+        PageRepository pageRepository = appContext.getBean(PageRepository.class);
+
+        Page page = pageRepository.findOne(new PageIdentity(language, pageId));
+
+        if(page == null) {
+            response.setStatus(HttpStatus.NOT_FOUND.value());
+            return;
+        }
+
+        byte[] bytesData;
+        try {
+            bytesData = data.getBytes();
```

Comment:
Would be good to do some checks of the uploaded file to verify it's not binary file or similar.

Reply:
@PascalLike added a file type check so this is resolved ?